File: src/test/java/com/radixpro/enigma/shared/RangeTest.java
/*
 * Jan Kampherbeek, (c) 2020.
 * Enigma is open source.
 * Please check the file copyright.txt in the root of the source for further details.
 */

package com.radixpro.enigma.shared;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class RangeTest {

   private static final double DELTA = 0.00000001;
   private Range range;


   @Before
   public void setUp() {
      range = new Range(100.0, 200.0);
   }

   @Test
   public void checkValueNoChange() {
      assertEquals(150.0, range.checkValue(150.0), DELTA);
   }

   @Test
   public void checkValueToLarge() {
      assertEquals(120.0, range.checkValue(220.0), DELTA);
   }

   @Test
   public void checkValueToSmall() {
      assertEquals(150.0, range.checkValue(50.0), DELTA);
   }

   @Test
   public void checkValueNegative() {
      assertEquals(150.0, range.checkValue(-50.0), DELTA);
   }

}

File: src/test/java/com/radixpro/enigma/be/util/RangeTest.kt
/*
 * Jan Kampherbeek, (c) 2020.
 * Enigma is open source.
 * Please check the file copyright.txt in the root of the source for further details.
 *
 */

package com.radixpro.enigma.be.util

import com.radixpro.enigma.testsupport.TestConstants.DELTA_8_POS
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

internal class RangeTest {

    @Test
    fun checkValueNoChance() {
        assertEquals(213.0, Range.checkValue(213.0), DELTA_8_POS)
    }

    @Test
    fun checkValueTooSmall() {
        assertEquals(350.0, Range.checkValue(-10.0), DELTA_8_POS)
    }

    @Test
    fun checkValueTooLarge() {
        assertEquals(50.0, Range.checkValue(410.0), DELTA_8_POS)
    }

    @Test
    fun checkValueWithExplicitRange() {
        assertEquals(33.3, Range.checkValue(33.3, 0.0, 100.0), DELTA_8_POS)
    }

    @Test
    fun checkValueWithWrongSequence() {
        assertEquals(33.3, Range.checkValue(33.3, 1000.0, 0.0), DELTA_8_POS)
    }

}